### PR TITLE
Support podman's `--preserve-fds` arg to container run/exec

### PR DESCRIPTION
**Pull request by LewisGaul**
Tested manually, but should probably add a testcase too... need to figure out the best way to do so.

- [x] Add testcase

### CODE REVIEW

File: python_on_whales/components/container/cli_wrapper.py [hunk 1]
```diff
@@ -860,6 +861,9 @@ def execute(
                 to allow communication with the parent process.
                 Currently only works with `tty=True` for interactive use
                 on the terminal.
+            preserve_fds: The number of additional file descriptors to pass
+                through to the container. Only supported by podman.
+
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:866]:
```suggestion
                through to the container. Only supported by podman.
```

File: python_on_whales/components/container/cli_wrapper.py [hunk 2]
```diff
@@ -1725,12 +1739,17 @@ def run(
                     "It's not possible to stream and detach a container at "
                     "the same time."
                 )
+
+        if preserve_fds:
+            pass_fds = range(3, 3 + preserve_fds)
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/container/cli_wrapper.py:1744]: I wouldn't mind an explanation in a code comment for this line. That should speed up the comprehension for future readers

File: tests/python_on_whales/components/test_container.py [hunk 3]
```diff
@@ -218,6 +218,17 @@ def test_create_with_systemd_mode(podman_client: DockerClient):
         assert container.config.systemd_mode is True
 
 
+def test_run_with_preserve_fds(podman_client: DockerClient):
+    read_fd, write_fd = os.pipe()
+    with podman_client.container.run(
+        "ubuntu",
+        ["bash", "-c", f"echo foobar >&{write_fd}"],
+        detach=True,
+        preserve_fds=write_fd - 2,
```
Comment (gabrieldemarmiesse) [inline tests/python_on_whales/components/test_container.py:227]: Same here

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]
Thanks for the pull request. I'm not super familiar with file descriptors. I guess we can assume that a reader knows they're represented by integers. We should just add a few comments here and there to explain the magic numbers and we should be good to go

File: python_on_whales/components/container/cli_wrapper.py (hunk 1 above)
Comment (gabrieldemarmiesse) [reply python_on_whales/components/container/cli_wrapper.py:866]: There is an empty extra line, this will break the docs formatting

Comment (LewisGaul) [reply python_on_whales/components/container/cli_wrapper.py:866]: oops

**Review by gabrieldemarmiesse — APPROVED** [review]
Looks good, a high quality PR once again :)

**Pull request merged**